src/Algebra/Properties/CommutativeMonoid.agda

```diff
@@ -0,0 +1,189 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some derivable properties
+------------------------------------------------------------------------
+
+open import Algebra
+
+module Algebra.Properties.CommutativeMonoid {g₁ g₂} (M : CommutativeMonoid g₁ g₂) where
+
+open import Algebra.Operations.CommutativeMonoid M
+open import Algebra.CommutativeMonoidSolver M
+open import Relation.Binary as B using (_Preserves_⟶_)
+open import Function
+open import Function.Equality using (_⟨$⟩_)
+open import Data.Product
+open import Data.Nat using (ℕ; zero; suc)
+open import Data.Fin using (Fin; punchIn; zero; suc)
+open import Data.List as List using ([]; _∷_)
+import Data.Fin.Properties as FP
+open import Data.Fin.Permutation as Perm using (Permutation; Permutation′; _⟨$⟩ˡ_; _⟨$⟩ʳ_)
+open import Data.Fin.Permutation.Components as PermC
+open import Data.Table as Table
+open import Data.Table.Relation.Equality as TE using (_≗_)
+open import Data.Unit using (tt)
+open import Data.Empty using (⊥-elim)
+import Data.Table.Properties as TP
+open import Relation.Binary.PropositionalEquality as P using (_≡_)
+open import Relation.Nullary using (yes; no)
+open import Relation.Nullary.Decidable using (⌊_⌋)
+
+open CommutativeMonoid M
+  renaming
+  ( ε to 0#
+  ; _∙_ to _+_
+  ; ∙-cong to +-cong
+  ; identity to +-identity
+  ; assoc to +-assoc
+  ; comm to +-comm
+  )
+open import Algebra.FunctionProperties _≈_
+open import Relation.Binary.EqReasoning setoid
+
+module _ {n} where
+  open B.Setoid (TE.setoid setoid n) public
+    using ()
+    renaming (_≈_ to _≋_)
+
+-- When summing over a function from a finite set, we can pull out any value and move it to the front.
+
+sumₜ-punchIn : ∀ {n} t (i : Fin (suc n)) → sumₜ t ≈ lookup t i + sumₜ (rearrange (punchIn i) t)
+sumₜ-punchIn f zero = refl
+sumₜ-punchIn {zero} t (suc ())
+sumₜ-punchIn {suc n} t (suc i) =
+  begin
```

Comment:
Can you align the arguments? It makes them much easier to read. If there are implicit arguments that are only used in certain cases, I usually put `{_}` when they're not needed to make it easier to parse.

Also in this particular case, you call the table `f` in the first case and then `t` in the second and third.